python/ray/node.py

```diff
@@ -96,10 +96,6 @@ def __init__(
             if len(external_redis) == 1:
                 external_redis.append(external_redis[0])
             [primary_redis_ip, port] = external_redis[0].split(":")
-            ray._private.services.wait_for_redis_to_start(
```

Comment:
Why do we remove `wait_for_redis_to_start()` here and in other places?

Reply:
1) The user should provide a Redis cluster and make sure it's up, so this logic doesn't fit here.

2) We have retry in gcs&raylet, so it should be ok.

Reply:
What are the remaining usages of `wait_for_redis_to_start()`, and can they be moved to a test-only util or removed as well?

Reply:
+1 to @rkooo567. I didn't find the right way to phrase it earlier, but I think giving a good error message if Redis is unavailable is important. Having a dedicated "wait and check Redis connection" logic seems the easiest way to achieve that.

Reply:
Since all the logic related to the storage are moved to GCS, we probably should just pass that information from GCS to the driver.

I don't see any benefit of detecting it here in the python script especially given that it'll need the user to `pip install redis` to use this.